### DISCUSSION TITLE
Captainhook and Docker documentation

### DISCUSSION THREAD

**Discussion by noteven** [captainhook-git/captainhook#246] [Q&A] [answered]
Currently using CaptainHook with docker and the following config (nesting of `run` and `mode` inside of `run` which is nested inside of `config` as per https://captainhookphp.github.io/captainhook/configure.html under `Settings`):

```json
{
    "config": {
        "fail-on-first-error": false,
        "run": {
            "mode": "docker",
            "exec": "docker run -i -v ${PWD}:/captainhook -w /captainhook api_dev:latest"
        }
    }
}
```

But I notice the CaptainHook `Install` docs (https://captainhookphp.github.io/captainhook/install.html) state that the config should use `run-mode` and `run-exec` inside of `config` directly like so:
```json
{
    "config": {
        "run-mode": "docker",
        "run-exec": "docker run -i -v ${PWD}:/captainhook -w /captainhook api_dev:latest"
    }
}
```

Are the two equivalent?

**Comment by sebastianfeldmann** [comment, marked as answer]
Both settings work.
The nested one is the currently preferred one but the prefixed `run-*` settings are still supported and get converted internally.
Possible that in a future major version I will remove the automatic conversion so updating your settings to the nested version is the right way.

**Reply by noteven** [reply]
Hello, already using the nested version, so luckily that's fine. Guess the `Install` docs should be updated so as to avoid new users using the `run-*` settings which will face deprecation.
